### PR TITLE
Fix Source Link cross-targeting targets

### LINKED ISSUE #33297: Add tests validating bundled Source Link

### PR DESCRIPTION
**Pull request by tmat**
`IsCrossTargetingBuild` is only defined in `.targets`, not in `.props`.

Only `Microsoft.SourceLink.Common.targets` needs to be imported in cross-targeting build as it contains workaround for https://github.com/Microsoft/msbuild/issues/3294. Once this issue is fixed we can remove cross-targeting specialization entirely.

Related: https://github.com/dotnet/msbuild/pull/8898
Fixes https://github.com/dotnet/sdk/issues/33297

### CODE REVIEW

**Comment by ViktorHofer** [comment]
Can you please add a test that protects the behavior for multi-targeting and non multi-targeting projects?

**Comment by tmat** [comment]
@ViktorHofer Will do that as a follow-up: https://github.com/dotnet/sdk/issues/33297

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.SourceLink.props [hunk 1]
```diff
@@ -13,20 +13,19 @@ Copyright (c) .NET Foundation. All rights reserved.
...
-    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.AzureRepos.Git\$(_SourceLinkSdkSubDir)\Microsoft.SourceLink.AzureRepos.Git.props"/>
-    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.Bitbucket.Git\$(_SourceLinkSdkSubDir)\Microsoft.SourceLink.Bitbucket.Git.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.Build.Tasks.Git\build\Microsoft.Build.Tasks.Git.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.Common\build\Microsoft.SourceLink.Common.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.GitHub\build\Microsoft.SourceLink.GitHub.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.GitLab\build\Microsoft.SourceLink.GitLab.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.AzureRepos.Git\build\Microsoft.SourceLink.AzureRepos.Git.props"/>
+    <Import Project="$(MSBuildThisFileDirectory)..\..\Microsoft.SourceLink.Bitbucket.Git\build\Microsoft.SourceLink.Bitbucket.Git.props"/>
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.SourceLink.props:28]: Are there any differences in these files between the `build` and `buildCrossTargeting` directories?

**Review by dsplaisted — APPROVED** [review]
Looks good I think.  Is there a reason not to add the tests with this PR as opposed to later?  It seems like that would give more confidence.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.SourceLink.props (hunk 1 above)
Comment (tmat) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.SourceLink.props:28]: The ones in buildCrossTargeting just import the ones in build.

**Comment by ViktorHofer** [comment]
Arcade is blocked anyway for about a month until the next (signed) preview that contains this fix releases.

**Comment by dsplaisted** [comment]
OK, go ahead and merge if you'd like.

**Comment by tmat** [comment]
Added a test - figured I can reuse the existing one, just add packaging validation.

**Pull request merged**